### PR TITLE
AP_BattMonitor: read SMBus SpecificationInfo() and update voltage and current multipliers

### PR DESCRIPTION
**Pull request by be-ska**
The SpecificationInfo() SMBus function is currently used only to check if the Smart Battery support PEC, based on the version number, but that function also returns voltage and current multipliers. This PR request add voltage and current multipliers when using "SMBus-Generic" battery monitor.
The current multiplier is particularly useful, because the int16_t representing current overflows at 32 768 mA, that may be not enough for medium size UAV.

SpecificationInfo
Current()

PS: this is my first PR, hope everything was done in the right way.

### CODE REVIEW

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.cpp [hunk 1]
```diff
@@ -147,31 +147,66 @@ bool AP_BattMonitor_SMBus_Generic::check_pec_support()
...
+            _i_multiplier = 1;
+            _specification_info_confirmed = true;
             return true;
         }
     }
+    // extract version
+    uint8_t version = (data & 0xF0) >> 4;
+
+    // version less than 0011b (i.e. 3) do not support PEC
+    if (version < 3) {
```
Comment (rmackay9) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.cpp:166]: Hi, thanks for this. I guess we could shorten this to a single line, "_pec_supported = version >= 3".  Alternatively just move the "else {" up one line to make it conform with AP's regular formatting style.

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.h [hunk 2]
```diff
@@ -23,11 +23,16 @@ class AP_BattMonitor_SMBus_Generic : public AP_BattMonitor_SMBus
...
+    // Override capacity scaler, use the current multiplier from SpecificationInfo()
+    uint16_t get_capacity_scaler() const override { return _i_multiplier; }
 
-    uint8_t _pec_confirmed; // count of the number of times PEC has been confirmed as working
+    // Read SpecificationInfo() function, check if PEC is supported and update voltage and current multipliers
+    // returns true once SpecificationInfo() is returned from the Smart Battery
+    bool read_specification_info();
+
+    bool _specification_info_confirmed = false; // specification info successfully read
+    uint16_t _v_multiplier;    // voltage multiplier = 10^VScale
```
Comment (rmackay9) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.h:34]: I wonder if we should consider initialising these to "1" so that the multiplier will return 1 even before the read_specification_info() method has been called.

**Comment by rmackay9** [comment]
Thanks very much for this.  I wonder what testing has been done.  In particular we should check that some of our existing SMBus batteries work correctly.  So for example the 3DR and Maxell batteries are two we should test if we can.  I actually have both although the Maxell is completely dead and I sadly don't have a charger for it immediately available.

**Comment by be-ska** [comment]
Hello @rmackay9, thank you for your reviews, I've committed the changes you suggested.
About the tests, I've only tested it with a SMBus Smart Battery I'm developing. In particular, I'm using a "current multiplier = 10" and the bench tests were successful.
Just for your reference I post a screen were both capacity and current are multiplied for the coefficient, so current can pass the 33 A limitation.
Unluckily I can't test other Smart Batteries, because I actually don't have them. I'm quite confident that the changes I've made shouldn't interfere with the other supported SMBus batteries, but I agree they should be tested.
QGC

**Comment by rmackay9** [comment]
@be-ska,

Great thanks.  I've added this to the EU Dev call (Wed at 4pm Japan time) so maybe we can find some other devs with SMBus batteries to help us test.

**Comment by tridge** [comment]
@be-ska nice! I've asked @hendjoshsr71 to take a look, he has a bunch of SMBus devices and may not the likelyhook or a broken implementation causing issues
